lib/timers.js

```diff
@@ -434,6 +482,15 @@ function rearm(timer) {
 
 
 const clearTimeout = exports.clearTimeout = function(timer) {
+  // Fewer checks may be possible, but these cover everything.
+  if (async_hook_fields[kDestroy] > 0 &&
+      timer &&
+      typeof timer[async_id_symbol] === 'number' &&
+      !timer._destroyed) {
+    emitDestroy(timer[async_id_symbol]);
+    timer._destroyed = true;
+  }
```

Comment:
The change is intentional. The previous implementation was fragile, so I decided it would be better to harden by restricting the scope to only timers created with `setImmediate`/`setTimeout`/`setInterval`. I'd prefer to add a test that shows how not having these calls in `unenroll` and `Timeout#close` causes the async graph to be incorrect before attempting to change their locations.